tests/realloc.c

```diff
@@ -15,6 +15,9 @@ Aborted
 int main()
 {
 	char *s = malloc((size_t)10);
-	realloc(s, (size_t)15);
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-result"
+	realloc(s, (size_t)15); // ignore return value. s might be invalidated by this call.
```

Comment:
Can't we just do `s = reallic(s, (size_t)15);` instead?

Reply:
I think the purpose of this test is to show that mishandling the result of `realloc` doesn't immediately trigger corruptions.

`crazyrealloc` triggers a double-free while reasonable `libc`s probably don't.

Note: My setup doesn't report the double free unless `MALLOC_CHECK_=2` is set.
```
⋊> env MALLOC_CHECK_=2 ltrace ./realloc
malloc(10)                                                       = 0x55d8a5798010
realloc(0x55d8a5798010, 15)                                      = 0x55d8a5798010
free(0x55d8a5798010)                                             = <void>
+++ exited (status 0) +++

⋊> env MALLOC_CHECK_=2 LD_PRELOAD=../src/crazyrealloc.so ltrace ./realloc
malloc(10)                                                       = 0x55cc19e88010
realloc(0x55cc19e88010, 15)                                      = 0x55cc19e88030
free(0x55cc19e88010free(): invalid pointer
 <no return ...>
--- SIGABRT (Aborted) ---
+++ killed by SIGABRT +++
```